src/Channels/ReadableBuffer.cs

```diff
@@ -573,6 +574,38 @@ internal static int FindFirstEqualByteSlow(ref Vector<byte> byteEquals)
         }
 
         /// <summary>
+        /// Reads a structure of type T out of a buffer of bytes.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public T Read<[Primitive]T>() where T : struct
+        {
+            var len = Unsafe.SizeOf<T>();
+            var span = FirstSpan;
+            return span.Length >= len ? span.Read<T>() : ReadMultiSpan<T>(len);
+        }
+        private unsafe T ReadMultiSpan<[Primitive]T>(int len) where T : struct
+        {
+            byte* local = stackalloc byte[len];
```

Comment:
Was trying to minimise overhead. Completely untested gut feel that this would be more efficient. Slice is simpler.